src/index.d.ts

```diff
@@ -0,0 +1,16 @@
+import { Component, FunctionalComponent } from 'preact';
+
+interface IAsyncRouteProps {
+    component?: any;
```

Comment:
this should be `JSX.Element`

Reply:
A `JSX.Element` is the output type, not the type of component.

If it was a `JSX.Element`, `h` would not need to be called on it: https://github.com/prateekbh/preact-async-route/blob/master/src/index.js#L58

In an ideal world, it would have the type `Component | FunctionalComponent`, however due to limitations in the current version of TypeScript, an untyped generic cannot itself be used as a type. (addressable by [this?](https://github.com/Microsoft/TypeScript/issues/1213)).